src/commands/force/user/create.ts

```diff
@@ -70,20 +71,25 @@ export class UserCreateCommand extends SfdxCommand {
     // because fields is type UserFields & Dictionary<string> we can access these
     const permsets: string = fields.permsets;
     const generatepassword: string = fields.generatepassword;
+    const profileName = fields.profileName;
 
     // extract the fields and then delete, createUser doesn't expect a permsets or generatepassword
     delete fields.permsets;
     delete fields.generatepassword;
     delete fields.generatePassword;
+    delete fields.profileName;
 
     try {
-      await this.user.createUser(fields);
+      this.authInfo = await this.user.createUser(fields);
+      fields.permsets = permsets;
```

Comment:
We delete them and then add them back on. Maybe we can clean this up and do something like:

```
this.authInfo = await this.user.createUser(stripInvalidAPIFIelds(fields));
```

The `stripInvalidAPIFIelds` would return a *copy* of the fields object just to be used for user creation. The original fields would be unmodified, and you don't have to delete then add them back on.

Reply:
Then you could also get rid of lines 71-74.